lightning/src/ln/channelmanager.rs

```diff
@@ -404,6 +404,11 @@ pub struct ChannelDetails {
 	pub short_channel_id: Option<u64>,
 	/// The node_id of our counterparty
 	pub remote_network_id: PublicKey,
+	/// The Features this node provided us opon our last connection with them.
+	/// This is particularly useful for routing as many node_announcement-context features are also
+	/// set in the init-context, and we should almost certainly consider the init-context version
+	/// to be the latest copy.
+	pub remote_node_features: InitFeatures,
```

Comment:
Can we reconcile the overloading of `node_features`? Here, it is referring to something of type `InitFeatures` but in `RouteHop` the `node_features` field has type `NodeFeatures`. More confusingly, they are used together in `Router::get_route`.

Reply:
Took your wording of "counterparty".